components/Home/Mech/Request.jsx

```diff
@@ -7,24 +7,35 @@ import RequestForm from 'common-util/List/RequestForm';
 import { AlertSuccess, AlertError } from 'common-util/List/ListCommon';
 import { getMechContract } from 'common-util/Contracts';
 import { HeaderTitle } from 'common-util/Title';
-import { FormContainer } from 'components/styles';
-import { getIpfsResponse } from 'common-util/functions';
+import { getAgentHash, getIpfsResponse } from 'common-util/functions';
+import { FormContainer } from '../../styles';
+import { getAgent } from '../Registry/utils';
 
 const Request = ({ account }) => {
   const [dataList, setDataList] = useState([]);
   const [error, setError] = useState(null);
   const [information, setInformation] = useState(null);
   const [isLoading, setIsLoading] = useState(false);
   const router = useRouter();
-  const { hash } = router.query;
+  const hash = router?.query?.hash;
 
   useEffect(() => {
-    const getData = async () => {
-      const data = await getIpfsResponse(hash);
+    const getIpfsDetailsFromHash = async (e) => {
+      const data = await getIpfsResponse(e);
       setDataList(data.tools || []);
     };
 
-    if (hash) getData();
+    const getIpfsDetailsFromId = async () => {
+      const agentDetails = await getAgent('3');
+      const currentHash = getAgentHash(agentDetails.agentHashes);
+      await getIpfsDetailsFromHash(currentHash);
+    };
+
+    if (hash) {
+      getIpfsDetailsFromHash(hash);
```

Comment:
Personally I don't like the complexity of supporting the hash like this. I can see that it saves a request, but I would prefer to simply have one way to do it for now – there is an address passed to the mech via the URL, then we get the hash from that address, and the data from IPFS from there.